plugins/chat/assets/javascripts/discourse/services/chat-composer-warnings-tracker.js

```diff
@@ -63,61 +59,43 @@ export default class ChatComposerWarningsTracker extends Service {
       return;
     }
 
-    const mentions = this._extractMentions(currentMessage.message);
-    this.mentionsCount = mentions?.length;
-
-    if (this.mentionsCount > 0) {
-      this.tooManyMentions =
-        this.mentionsCount > this.siteSettings.max_mentions_per_chat_message;
+    currentMessage.cook().then(() => {
+      const mentions = parseMentionedUsernames(currentMessage.cooked);
```

Comment:
There are only two changes in this method:
1. Now we call currentMessage.cook() before parsing mentions
2. We call `parseMentionedUsernames(currentMessage.cooked)` instead of `this._extractMentions(currentMessage.message)`